org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandler.java

```diff
@@ -77,8 +86,27 @@
 		if (offset >-1 && !monitor.isCanceled()) {
 			IBuffer buffer = unit.getBuffer();
 			if (buffer != null && buffer.getLength() >= offset) {
-				unit.codeComplete(offset, collector, monitor);
-				proposals.addAll(collector.getCompletionItems());
+				IProgressMonitor subMonitor = new NullProgressMonitor() {
```

Comment:
can't you wrap the outer monitor so that if the user cancels it, the submonitor is cancelled too?